man/sd_notify.xml

```diff
@@ -312,13 +311,14 @@
   <refsect1>
     <title>Return Value</title>
 
-    <para>On failure, these calls return a negative errno-style error
-    code. If <varname>$NOTIFY_SOCKET</varname> was not set and hence
-    no status data could be sent, 0 is returned. If the status was
-    sent, these functions return with a positive return value. In
-    order to support both, init systems that implement this scheme and
-    those which do not, it is generally recommended to ignore the
-    return value of this call.</para>
+    <para>On failure, these calls return a negative errno-style error code. If <varname>$NOTIFY_SOCKET</varname> was
+    not set and hence no status data could be sent, 0 is returned. If the status was sent, these functions return with
+    a positive return value. In order to support both service managers that implement this scheme and those which do
+    not, it is generally recommended to ignore the return value of this call. Note that the return value simply
+    indicates whether the notification message was enqueued properly, it does not reflect whether the message could be
+    stored successfully. Specifically, no error is returned when a file descriptor is attempted to be stored using
+    <varname>FDSTORE=1</varname> but the service is not actually configured to permit storing of file
+    descriptors.</para>
```

Comment:
That's a bugger. Maybe we should allow querying or actually return some code if storing of the fds fails. Subject for another PR though.